VideoApp/VideoApp/Stores/Chat/ChatStore.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  Copyright (C) 2021 Twilio, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import TwilioConversationsClient
+
+protocol ChatStoreWriting: AnyObject {
+    var connectionState: ChatConnectionState { get }
+    var messages: [TCHMessage] { get }
+    func connect(accessToken: String, conversationName: String)
+    func disconnect()
+}
+
+class ChatStore: NSObject, ChatStoreWriting {
+    enum Update {
+        case didChangeConnectionState
+    }
+
+    static let shared: ChatStoreWriting = ChatStore()
```

Comment:
I'm still thinking about if I'm going to keep the singleton. I'm leaning towards keeping it because really we shouldn't be manually passing dependencies like this from screen to screen. It doesn't scale. This is inline with our [dependency injection strategy for this app](https://docs.google.com/document/d/1NE1Ou8J9lVpYoeTe9Xxe0o-BwyeIIoyJe2_my2KKH-A/edit?usp=sharing). It's better than using a DI framework that would make it more difficult for customers to reuse code.

I mean we may get away with passing between just 2 screens here for now but may run into issues down the road. I originally thought that I was going to have to call `leave` on the conversation when done and that is async so `ChatStore` really would need to live forever. But I don't have to call `leave` because our chat feature is so simple.